src/metabase/driver/util.clj

```diff
@@ -248,7 +248,10 @@
         false))))
 
 (def ^:private memoized-supports?*
-  (mdb/memoize-for-application-db supports?*))
+  (memoize/memo
+   (-> supports?*
+       (vary-meta assoc ::memoize/args-fn (fn [[driver feature _database]]
+                                            [driver feature (mdb/unique-identifier)])))))
```

Comment:
Sadly `mdb/unique-identifier` isn't going to cut it because it is a unique identifier for the app DB. `mdb/unique-identifier` will be the same across all databases. `database` is an entity that represents a connected "customer" DB. 

If we need something that is cheap to hash and correctly invalidates the `supports?` function I would suggest using a combination of `(mdb/unique-identifier)`, `(:id database)`, and `(:updated_at database)`. `(:updated_at database)` will change at the same rate the hash of `database` changes, so this will preserve existing behaviour.
```suggestion
                                            [driver feature (mdb/unique-identifier) (:id database) (:updated_at database)])))))
```

Reply:
Gotcha, my bad! Thank you for suggesting a fix. I'm still getting confused between the application database and the processed databases.